src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -324,8 +324,9 @@ public void OnFinishEditing()
             editedSpecies.FormattedName);
 
         // Update name
+        NewName = gui.SpeciesNameEdit.Text;
```

Comment:
I really think this should be `gui.GetNewSpeciesName()` so that this class doesn't touch the internals of the GUI as that's dirty.